image-download

```diff
@@ -292,10 +292,16 @@ def download_images(image_list, force, quiet, state, store):
                 target = committed_target(image)
 
             # don't download the same thing multiple times in parallel
-            wait_lock(target)
+            lockfile = wait_lock(target)
 
             if force or state or not os.path.exists(target):
                 download(target, force, state, quiet, store)
+
+            # clean up lock file; parallel runs may remove it already
+            try:
+                os.unlink(lockfile)
```

Comment:
Okay, so the problem here is that unlinking lockfiles isn't really safe: someone might have just created the file again, and then you `unlink()` it while they think they still hold an fd that refers to it...

In practice, though, eh.... this is gonna happen *so* incredibly rarely, and this patch is a massive improvement compared to the current situation.